### ISSUE TITLE
implement `graceful shutdown` logic

### ISSUE THREAD

**Issue by konradkonrad** [raiden-network/raiden#77]
There seem to be three modes of shutdown we want to support:
- shutdown and leave: needs a "settle all" and then just shutdown
- ~shutdown and keep channels open "3rd party challenging"~
- unintentional shutdown and resync #69 #2 

For cleanly shutting down a node, there needs to be a shutdown process, that:
- [ ] settles open channels *or* wait for all pending transfers to complete

**Comment by LefterisJP** [comment]
Should this also not be part of the MVP milestone?

**Comment by heikoheiko** [comment]
ToDo: List of things that would be beneficial to do when there is a chance to do a controlled shut down

**Comment by konradkonrad** [comment]
depends on #613

**Comment by LefterisJP** [comment]
@konradkonrad With https://github.com/raiden-network/raiden/pull/599 this should be finished. If there is anything else remaining please create another issue for it.
